File: app/src/main/java/com/badradio/nz/utilities/Log.java
package com.badradio.nz.utilities;

public class Log {

    public static final boolean LOG = true;

    public static void i(String tag, String string) {
        if (LOG) android.util.Log.i(tag, string);
    }

    public static void e(String tag, String string) {
        if (LOG) android.util.Log.e(tag, string);
    }

    public static void e(String tag, String string, Exception e) {
        if (LOG) android.util.Log.e(tag, string, e);
    }

    public static void d(String tag, String string) {
        if (LOG) android.util.Log.d(tag, string);
    }

    public static void v(String tag, String string) {
        if (LOG) android.util.Log.v(tag, string);
    }

    public static void w(String tag, String string) {
        if (LOG) android.util.Log.w(tag, string);
    }

    public static void w(String tag, String string, Exception e) {
        if (LOG) android.util.Log.w(tag, string, e);
    }

    public static void printStackTrace(Exception e) {
        if (LOG) e.printStackTrace();
    }

}

File: app/src/main/java/com/badradio/nz/utilities/Log.kt
package com.badradio.nz.utilities

import android.util.Log
import com.badradio.nz.Config.LOG
import java.lang.Exception

object Log {
    fun i(tag: String, string: String) {
        if (LOG) Log.i(tag, string)
    }

    fun e(tag: String, string: String) {
        if (LOG) Log.e(tag, string)
    }

    @JvmStatic
    fun e(tag: String, string: String, e: Exception) {
        if (LOG) Log.e(tag, string, e)
    }

    fun d(tag: String, string: String) {
        if (LOG) Log.d(tag, string)
    }

    @JvmStatic
    fun v(tag: String, string: String) {
        if (LOG) Log.v(tag, string)
    }

    fun w(tag: String, string: String) {
        if (LOG) Log.w(tag, string)
    }

    fun w(tag: String, string: String, e: Exception) {
        if (LOG) Log.w(tag, string, e)
    }

    @JvmStatic
    fun printStackTrace(e: Exception) {
        if (LOG) e.printStackTrace()
    }
}